homeassistant/components/yale/sensor.py

```diff
@@ -0,0 +1,210 @@
+    """Mixin for required keys."""
+
+    value_fn: Callable[[_T], int | None]
+
+
+SENSOR_TYPE_DEVICE_BATTERY = YaleSensorEntityDescription[LockDetail](
+    key="device_battery",
+    entity_category=EntityCategory.DIAGNOSTIC,
+    state_class=SensorStateClass.MEASUREMENT,
+    value_fn=_retrieve_device_battery_state,
+)
+
+SENSOR_TYPE_KEYPAD_BATTERY = YaleSensorEntityDescription[KeypadDetail](
+    key="linked_keypad_battery",
+    entity_category=EntityCategory.DIAGNOSTIC,
+    state_class=SensorStateClass.MEASUREMENT,
+    value_fn=_retrieve_linked_keypad_battery_state,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: YaleConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Yale sensors."""
+    data = config_entry.runtime_data
+    entities: list[SensorEntity] = []
+
+    for device in data.locks:
+        detail = data.get_device_detail(device.device_id)
+        entities.append(YaleOperatorSensor(data, device, "lock_operator"))
+        if SENSOR_TYPE_DEVICE_BATTERY.value_fn(detail):
+            entities.append(
+                YaleBatterySensor[LockDetail](data, device, SENSOR_TYPE_DEVICE_BATTERY)
+            )
+        if keypad := detail.keypad:
+            entities.append(
+                YaleBatterySensor[KeypadDetail](
+                    data, keypad, SENSOR_TYPE_KEYPAD_BATTERY
+                )
+            )
+
+    entities.extend(
+        YaleBatterySensor[Doorbell](data, device, SENSOR_TYPE_DEVICE_BATTERY)
+        for device in data.doorbells
+        if SENSOR_TYPE_DEVICE_BATTERY.value_fn(data.get_device_detail(device.device_id))
+    )
+
+    async_add_entities(entities)
+
+
+class YaleOperatorSensor(YaleEntityMixin, RestoreSensor):
+    """Representation of an Yale lock operation sensor."""
+
+    _attr_translation_key = "operator"
+    _operated_remote: bool | None = None
+    _operated_keypad: bool | None = None
+    _operated_manual: bool | None = None
+    _operated_tag: bool | None = None
+    _operated_autorelock: bool | None = None
+
+    @callback
+    def _update_from_data(self) -> None:
+        """Get the latest state of the sensor and update activity."""
+        self._attr_available = True
+        if lock_activity := self._get_latest({ActivityType.LOCK_OPERATION}):
+            lock_activity = cast(LockOperationActivity, lock_activity)
+            self._attr_native_value = lock_activity.operated_by
+            self._operated_remote = lock_activity.operated_remote
+            self._operated_keypad = lock_activity.operated_keypad
+            self._operated_manual = lock_activity.operated_manual
+            self._operated_tag = lock_activity.operated_tag
+            self._operated_autorelock = lock_activity.operated_autorelock
+            self._attr_entity_picture = lock_activity.operator_thumbnail_url
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
```

Comment:
How much do users notice about this migration? Is it worth trying to split these attributes off directly so we don't have to deprecate them at some point?

Reply:
Long term these should be removed but I know users have these in automations so its something we need to give them notice before we migrate them since we are already putting them though enough pain with a transition and I'd like to change both yale and august at the same time.